discovery-provider/src/tasks/entity_manager/developer_app.py

```diff
@@ -1,31 +1,55 @@
 import json
+from datetime import datetime
 from typing import Optional, TypedDict, Union, cast
 
+from eth_account.messages import defunct_hash_message
 from src.models.grants.developer_app import DeveloperApp
 from src.tasks.entity_manager.utils import (
     Action,
     EntityType,
     ManageEntityParameters,
     copy_record,
 )
+from src.utils import web3_provider
 from src.utils.indexing_errors import EntityMissingRequiredFieldError
 from src.utils.model_nullable_validator import all_required_fields_present
 from src.utils.structured_logger import StructuredLogger
 
 logger = StructuredLogger(__name__)
 
 
+class AppSignature(TypedDict):
+    message: str
+    signature: str
+
+
 class CreateDeveloperAppMetadata(TypedDict):
-    address: Union[str, None]
     name: Union[str, None]
     description: Union[str, None]
     is_personal_access: Union[bool, None]
+    app_signature: AppSignature
 
 
 class RevokeDeveloperAppMetadata(TypedDict):
     address: Union[str, None]
 
 
+def get_app_address_from_signature(app_signature: AppSignature):
+    web3 = web3_provider.get_eth_web3()
+    message_hash = defunct_hash_message(text=app_signature["message"])
+    app_address = web3.eth.account.recoverHash(
+        message_hash, signature=app_signature["signature"]
+    )
+    return app_address.lower()
+
+
+def is_within_6_hours(timestamp_str):
+    current_time = datetime.now()
```

Comment:
i think we want to avoid using datetime.now() for something like this because it depends on the server's timing configuration

Reply:
ooo ty :o will look into using just a unix timestamp

Reply:
just wanna make sure the timing is accurate, what's the schema of the app signature message too?

Reply:
Yee so the message just looks like
`Creating Audius developer app at [timestamp]`

the timestamp and 6 hour check is more to help ensure bad actors don't reuse the signature, but its probably overkill. we don't actually use the timestamp outside of the check here.

it's a similar pattern to the discovery/identity API signature messages which are smthing like "Click sign to authenticate with Discover Provider: [timestamp]"

hope this makes sense, down to huddle if it doesn't!

Reply:
@raymondjacobson just making sure if we wanna do this since you have thoughts about using timestamp

Reply:
merging to unblock some other work but happy to get rid of the timestamp in another pr